src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -439,6 +439,21 @@ export function isLocalLaunch(configuration: IConfigurationService) {
     return false;
 }
 
+export function getInterpreterHashInMetdata(
+    notebookMetadata: nbformat.INotebookMetadata | undefined
+): string | undefined {
+    // If the user has kernelspec in metadata & the interpreter hash is stored in metadata, then its a perfect match.
+    // This is the preferred approach https://github.com/microsoft/vscode-jupyter/issues/5612
+    if (
+        typeof notebookMetadata === 'object' &&
+        'interpreter' in notebookMetadata &&
+        (notebookMetadata as any).interpreter &&
+        typeof (notebookMetadata as any).interpreter === 'object' &&
+        'hash' in (notebookMetadata as any).interpreter
+    ) {
+        return (notebookMetadata as any).interpreter.hash;
+    }
+}
```

Comment:
An unnecessary change, but will be required later.
Had this as part of some other perf fixes that I'm not submitting just yet, basically moved the extraction of interpreterHash from notebook metadata into a function.